scripts/get-workspace-map.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# get-workspace-map.sh
+#   Read Namespace resources and generate a JSON object mapping from namespaces
+#   to workspace based on the value of label toolchain.dev.openshift.com/space.
+#   This script assumes `oc` is preconfigured with all the required clusters
+#   and could be found in $PATH.
+#
+set -o pipefail -o errexit -o nounset
+
+# ======= Parameters ======
+# The following variables can be set from outside the script by setting
+# the same environment variables.
+#
+# If set to "true", query all staging member clusters for which contexts exist locally
+QUERY_STAGE="${QUERY_STAGE:-true}"
+# If set to "true", query all production member clusters for which contexts exist locally
+QUERY_PROD="${QUERY_PROD:-""}"
+# List of space-separated locally-defined context names to be queried
+QUERY_URLS="${QUERY_URLS:-""}"
+
+STAGE=""
+PROD=""
+
+if [[ $QUERY_STAGE == "true" ]]; then
+# Take all non-host staging clusters and keep only their names
+  STAGE=$(
+    oc config get-contexts \
+      | { grep -E 'api-stone-stg\S+' || test $? = 1; } \
+      | { grep -v api-stone-stg-host || test $? = 1; } \
+      | sed 's/^[ \*]*//g' \
+      | sed 's/ .*$//g' \
+  )
+  if [[ -z $STAGE ]]; then
+    echo "No staging clusters found."
+  fi
+fi
+
+if [[ $QUERY_PROD == "true" ]]; then
+# Take all non-host production clusters and keep only their names
+  PROD=$(
+    oc config get-contexts \
+      | { grep -E 'api-stone-prd\S+' || test $? = 1; } \
+      | { grep -v api-stone-prd-host || test $? = 1; } \
+      | sed 's/^[ \*]*//g' \
+      | sed 's/ .*$//g' \
+  )
+  if [[ -z $PROD ]]; then
+    echo "No production clusters found."
+  fi
+fi
+
+if [[ -z $STAGE && -z $PROD && -z $QUERY_URLS ]]; then
+  echo "No clusters found."
+  exit 1
+fi
+
+contexts="${STAGE} ${PROD} ${QUERY_URLS}"
+
+out={}
+for context in $contexts
+do
+  echo "Querying cluster with context: $context..."
```

Comment:
The script currently returns malformed JSON due to this line. Please remove it.

Reply:
oh yes

Reply:
done